src/graphs.jl

```diff
@@ -83,3 +83,130 @@ end
     # markersize := node_weight
     dim==3 ? (x, y, z) : (x, y)
 end
+
+# ---------------------------------------------------------------------------
+# Arc Diagram
+
+function get_source_destiny_weight{T}(mat::AbstractArray{T,2})
+    nrow, ncol = size(mat)    # rows are sources and columns are destinies
+
+    nosymmetric = !issym(mat) # plots only triu for symmetric matrices
+    nosparse = !issparse(mat) # doesn't plot zeros from a sparse matrix
+
+    L = length(mat)
+
+    source  = Array(Int, L)
+    destiny = Array(Int, L)
+    weight  = Array(T, L)
+
+    idx = 1
+    for i in 1:nrow, j in 1:ncol
+        value = mat[i, j]
+        if !isnan(value) && ( nosparse || value != zero(T) ) # TODO: deal with Nullable
+
+            if i < j
+                source[idx]  = i
+                destiny[idx] = j
+                weight[idx]  = value
+                idx += 1
+            elseif nosymmetric && (i > j)
+                source[idx]  = i
+                destiny[idx] = j
+                weight[idx]  = value
+                idx += 1
+            end
+
+        end
+    end
+
+    resize!(source, idx-1), resize!(destiny, idx-1), resize!(weight, idx-1)
+end
+
+function get_source_destiny_weight(source::AbstractVector, destiny::AbstractVector)
+    if length(source) != length(destiny)
+        throw(ArgumentError("Source and destiny must have the same length."))
+    end
+    source, destiny, Float64[ 1.0 for i in source ]
+end
+
+function get_source_destiny_weight(source::AbstractVector, destiny::AbstractVector, weight::AbstractVector)
```

Comment:
I'd probably change this to:

```
function get_source_destiny_weight(source::AbstractVector, destiny::AbstractVector, weight::AbstractVector = ones(length(source)))
```

and remove the method above it.

I would also consider if this can't be represented equally well as a sparse matrix (which can be constructed with your source/destiny/weight vectors).  If so then we can support AbstractMatrix and be done.